### PR TITLE
Add tapir schemas for json filters

### PR DESCRIPTION
**Pull request by fazzou**
- Expose raw controller actions underlying doFilter and filterDefinition
- Require schemas for json filter fields
- Create CompoundResult aggregating filter definition and filter results
- Expose schemas for FilterDefinition and CompoundResult

### CODE REVIEW

File: src/test/scala/org/virtuslab/beholder/UserMachinesView.scala [hunk 1]
```diff
@@ -22,6 +23,8 @@ trait UserMachinesViewComponent
 
   import unicorn.profile.api._
 
+  implicit val dateSchema: Schema[Date] = Schema.string[Date]
```
Comment (mszturo) [inline src/test/scala/org/virtuslab/beholder/UserMachinesView.scala:26]: No format here?

File: src/main/scala/org/virtuslab/beholder/filters/json/JsonFilterFieldsComponent.scala [hunk 2]
```diff
@@ -169,23 +184,22 @@ trait JsonFilterFieldsComponent extends FilterFieldComponent with SeqParametersH
         override protected def filterOnColumn(column: Rep[T#Value])(dataSeq: Seq[T#Value]): Rep[Option[Boolean]] = {
           SeqParametersHelper.isColumnValueInsideSeq(column)(dataSeq)((column, data) => column.? === data)
         }
+        def filterSchema: Schema[Seq[T#Value]] = Schema.schemaForIterable
       }
     }
 
-    private implicit def rangeFormat[T: Format]: Format[FilterRange[T]] =
```
Comment (mszturo) [inline src/main/scala/org/virtuslab/beholder/filters/json/JsonFilterFieldsComponent.scala:175]: Is this implicit defined somewhere else in scope?

Comment (fazzou) [reply src/main/scala/org/virtuslab/beholder/filters/json/JsonFilterFieldsComponent.scala:175]: yes, moved it to `FilterRange` Companion object

File: src/test/scala/org/virtuslab/beholder/UserMachinesView.scala (hunk 1 above)
Comment (fazzou) [reply src/test/scala/org/virtuslab/beholder/UserMachinesView.scala:26]: turns out `Format[Date]` is availabile straight from play-json

Comment (fazzou) [reply src/test/scala/org/virtuslab/beholder/UserMachinesView.scala]: ah missed that you meant `.format` on `Schema`, added

**Review by mszturo — APPROVED** [review]

**Pull request merged**